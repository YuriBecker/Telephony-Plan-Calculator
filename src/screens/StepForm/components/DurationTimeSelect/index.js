import React from 'react';
import InputSpinner from 'react-native-input-spinner';
import { Text } from '@ui-kitten/components';
import PropTypes from 'prop-types';
import colors from '../../../../styles/colors';

const index = ({ value, setValue }) => {
  return (
    <>
      <Text
        category="h5"
        style={{
          textAlign: 'center',
          marginVertical: 15,
          color: colors.primary,
        }}
      >
        Quantos minutos de ligação?
      </Text>
      <InputSpinner
        max={999}
        step={1}
        colorMax={colors.primary}
        colorMin={colors.primary}
        value={value}
        onChange={setValue}
        color={colors.primary}
        inputStyle={{ color: colors.primary }}
        fontSize={18}
      />
    </>
  );
};

export default index;

index.propTypes = {
  value: PropTypes.number.isRequired,
  setValue: PropTypes.func.isRequired,
};
