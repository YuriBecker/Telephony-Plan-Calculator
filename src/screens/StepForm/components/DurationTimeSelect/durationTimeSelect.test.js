import React from 'react';
import { shallow } from 'enzyme';
import DurationTimeSelect from './index';

describe('Testing DurationTimeSelect component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<DurationTimeSelect value={1} setValue={jest.fn()} />);
  });
  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render the  Input Spinner', () => {
    expect(
      // wrapper.findWhere(node => node.prop('testID') === 'title')
      wrapper.find('InputSpinner')
    ).toExist();
  });

  it('should render the  title Text', () => {
    expect(
      // wrapper.findWhere(node => node.prop('testID') === 'title')
      wrapper.find('TextComponent')
    ).toExist();
  });
});
