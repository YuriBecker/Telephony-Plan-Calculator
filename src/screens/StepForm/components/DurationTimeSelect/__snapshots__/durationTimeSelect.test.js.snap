// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing DurationTimeSelect component should render correctly 1`] = `
<Fragment>
  <TextComponent
    category="h5"
    style={
      Object {
        "color": "#1db954",
        "marginVertical": 15,
        "textAlign": "center",
      }
    }
  >
    Quantos minutos de ligação?
  </TextComponent>
  <InputSpinner
    activeOpacity={0.85}
    arrows={false}
    background="transparent"
    buttonFontFamily={null}
    buttonFontSize={25}
    buttonLeftDisabled={false}
    buttonLeftText={null}
    buttonPressStyle={Object {}}
    buttonPressTextColor="#FFFFFF"
    buttonRightDisabled={false}
    buttonRightText={null}
    buttonStyle={Object {}}
    buttonTextColor="#FFFFFF"
    color="#1db954"
    colorLeft="#3E525F"
    colorMax="#1db954"
    colorMin="#1db954"
    colorPress="#3E525F"
    colorRight="#3E525F"
    disabled={false}
    editable={true}
    fontFamily={null}
    fontSize={18}
    height={50}
    inputStyle={
      Object {
        "color": "#1db954",
      }
    }
    max={999}
    min={0}
    onChange={[MockFunction]}
    precision={2}
    rounded={true}
    showBorder={false}
    step={1}
    style={Object {}}
    textColor="#000000"
    type="int"
    value={1}
    width={150}
  />
</Fragment>
`;
